fdbclient/include/fdbclient/BulkLoading.h

```diff
@@ -0,0 +1,203 @@
+/*
+ * BulkLoading.h
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2013-2024 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef FDBCLIENT_BULKLOADING_H
+#define FDBCLIENT_BULKLOADING_H
+#pragma once
+
+#include "fdbclient/FDBTypes.h"
+#include "fdbrpc/fdbrpc.h"
+
+enum class BulkLoadPhase : uint8_t {
+	Invalid = 0, // Set by users
+	Triggered = 1, // Update when DD trigger a data move for the task
+	Running = 2, // Update atomically with updating KeyServer dest servers in startMoveKey
+	Complete = 3, // Update atomically with updating KeyServer src servers in finishMoveKey
```

Comment:
Does the 'Complete' state exist for ever? Until an overwrite of the task key range?

Reply:
A task with 'Complete' state can be overwritten by a brand new task triggered by users.
A task with 'Complete' state can be cleared by a bulk load metadata cleanup (this implementation will be delivered by following PRs).

Reply:
ok